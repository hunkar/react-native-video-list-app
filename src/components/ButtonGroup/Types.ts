import {IconNode} from 'react-native-elements/dist/icons/Icon';
import {StyleProp, ViewStyle} from 'react-native';

//TODO Other styles can be customized. Can be added in the future.
/**
 * title: text of button
 * icon: icon of button
 * onPress: press event of button
 */
export type ButtonProps = {
  title: string;
  icon?: IconNode;
  onPress?: (() => void) | undefined;
};

/**
 * containerStyle: Styles for group container
 * buttons: array of ButtonProps
 */
export type ButtonGroupProps = {
  containerStyle?: StyleProp<ViewStyle>;
  buttons: ButtonProps[];
};
