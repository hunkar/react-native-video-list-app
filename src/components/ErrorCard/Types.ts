/**
 * title: Text of error.
 * onRetry: function that will be called on retry button clicked.
 */
export type ErrorCardProps = {
  title: string;
  onRetry?: () => void;
};
