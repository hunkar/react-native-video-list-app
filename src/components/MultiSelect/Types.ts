/**
 * id: id of item,
 * name: item name that will be shown
 * isSelected: you can set select status of item
 * value: extra value for item. For external use.
 */
export type MultiSelectItemProps = {
  id: string;
  name: string;
  isSelected?: boolean;
  value?: string | number | boolean;
};

/**
 * items: array of MultiSelectItemProps
 * selectedItems: id array of selected items
 * onChange: trigger function with selected items
 */
export type MultiSelectProps = {
  items: MultiSelectItemProps[];
  selectedItems?: string[];
  onChange?: (item: MultiSelectItemProps[]) => void;
};
