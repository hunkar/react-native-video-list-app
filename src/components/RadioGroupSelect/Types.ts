/**
 * id: item id
 * name: item name that will be show on the list
 * isSelected: item can be selectable with this prop
 * value: any value for external use.
 */
export type RadioGroupSelectItemProps = {
  id: string;
  name: string;
  isSelected?: boolean;
  value?: string | number | boolean;
};

/**
 * items: array of RadioGroupSelectItemProps items
 * selectedItem: id of selected item
 * onChange: trigger with selected item on change
 */
export type RadioGroupSelectProps = {
  items: RadioGroupSelectItemProps[];
  selectedItem?: string;
  onChange?: (item: RadioGroupSelectItemProps) => void;
};
