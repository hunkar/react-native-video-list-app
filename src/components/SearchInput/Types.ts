/**
 * placeholder: placeholder for searchBar
 * onChangeText: event for text change
 * changeTimeout: letter input frequency waiting range in ms 
 */
export type SearchInputProps = {
  placeholder?: string;
  onChangeText?: (text: string) => void;
  changeTimeout?: number;
};
