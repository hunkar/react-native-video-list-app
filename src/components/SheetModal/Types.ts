/**
 * title: title of modal
 * children: any ReactNode for modal child
 */
export type SheetModalProps = {
  title: string;
  children: React.ReactNode;
};
