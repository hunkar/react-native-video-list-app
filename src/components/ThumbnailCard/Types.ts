/**
 * id: Id of music.
 * title: Main title for singer name.
 * subTitle: Sub title for music name.
 * imageUrl: Url of video image.
 */
export type ThumbnailCardProps = {
  id: string;
  title: string;
  subTitle: string;
  imageUrl: string;
};
