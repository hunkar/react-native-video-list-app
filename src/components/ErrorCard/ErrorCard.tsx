import React from 'react';
import {StyleSheet, View} from 'react-native';
import {Card, Icon, Text} from 'react-native-elements';
import {ErrorCardProps} from './Types';

/**
 * This component shows message in error panel.
 * retryIcon is available if onRetry exists. This icon is for retry request or process if needed.
 * 
 * @param props ErrorCardProps
 * @returns ReactNode
 */
const ErrorCard = (props: ErrorCardProps) => (
  <Card containerStyle={styles.errorPanelCard}>
    <View style={styles.errorPanelContainer}>
      <Icon name="warning" color={'#ffa29c'} tvParallaxProperties={undefined} />
      <Text testID="title" style={styles.errorText}>
        {props.title}
      </Text>

      {props.onRetry && (
        <Icon
          testID="retryIcon"
          name="sync"
          color={'#ffa29c'}
          onPress={props.onRetry}
          tvParallaxProperties={undefined}
          containerStyle={styles.retryIconContiner}
        />
      )}
    </View>
  </Card>
);

const styles = StyleSheet.create({
  errorPanelCard: {
    borderColor: '#ffa29c',
    borderWidth: 1,
  },
  errorPanelContainer: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'flex-start',
    alignItems: 'center',
  },
  errorText: {
    color: '#ffa29c',
    marginLeft: 15,
  },
  retryIconContiner: {
    position: 'absolute',
    right: 0,
    borderColor: '#ffebeb',
    borderWidth: 1,
    padding: 5,
    borderRadius: 5,
  },
});

export default ErrorCard;
