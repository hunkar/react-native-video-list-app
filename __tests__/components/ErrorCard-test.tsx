import renderer, {act} from 'react-test-renderer';
import React from 'react';
import {ErrorCardProps} from '../../src/components/ErrorCard/Types';
import ErrorCard from '../../src/components/ErrorCard/ErrorCard';

test('snapshot equility', async () => {
  const mockProps: ErrorCardProps = {
    title: 'No data!',
  };

  const component = renderer.create(<ErrorCard {...mockProps} />);

  expect(component.toJSON()).toMatchSnapshot();
});

test('renders title correctly, no retry icon.', async () => {
  const mockProps: ErrorCardProps = {
    title: 'No data!',
  };

  const component = renderer.create(<ErrorCard {...mockProps} />);

  const titleComponent = component.root.findByProps({testID: 'title'}).props;

  const retryIconComponent = component.root.findAllByProps({
    testID: 'retryIcon',
  });

  expect(titleComponent.children).toEqual(mockProps.title);
  expect(retryIconComponent.length).toEqual(0);
});

test('renders title correctly, with icon.', async () => {
  const mockProps: ErrorCardProps = {
    title: 'No data!',
    onRetry: jest.fn(),
  };

  const component = renderer.create(<ErrorCard {...mockProps} />);

  const titleComponent = component.root.findByProps({testID: 'title'}).props;

  const retryIconComponent = component.root.findByProps({
    testID: 'retryIcon',
  }).props;

  expect(titleComponent.children).toEqual(mockProps.title);
  expect(retryIconComponent.onPress).toEqual(mockProps.onRetry);
});

test('renders title correctly, with icon.', async () => {
  const mockProps: ErrorCardProps = {
    title: 'No data!',
    onRetry: jest.fn(),
  };

  const component = renderer.create(<ErrorCard {...mockProps} />);

  const retryIconComponent = component.root.findByProps({
    testID: 'retryIcon',
  }).props;

  act(() => {
    retryIconComponent.onPress();
  });

  expect(retryIconComponent.onPress).toBeCalledTimes(1);
});
