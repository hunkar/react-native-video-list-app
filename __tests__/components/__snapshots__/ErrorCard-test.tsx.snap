// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot equility 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#ffa29c",
      "borderWidth": 1,
      "margin": 15,
      "marginBottom": 0,
      "padding": 15,
      "shadowColor": "rgba(0,0,0, .2)",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "flex-start",
        }
      }
    >
      <View
        style={
          Object {
            "overflow": "hidden",
          }
        }
      >
        <View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#ffa29c",
                    "fontSize": 24,
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="iconIcon"
            >
              î€‚
            </Text>
          </View>
        </View>
      </View>
      <Text
        style={
          Object {
            "color": "#ffa29c",
            "marginLeft": 15,
          }
        }
        testID="title"
      >
        No data!
      </Text>
    </View>
  </View>
</View>
`;
